### DISCUSSION TITLE
Правильно ли использовать вложенные слайсы?

### DISCUSSION THREAD

**Discussion by stsiushkevich** [feature-sliced/documentation#613] [Q&A] [answered]
Здравствуйте! Допустим мы имеем сущность Client, у которой есть документы, события, платежи, чеки и т.д. Правильно ли организовывать подслайсы? Например:

entities/Client/Client/ui/ClientList
entities/Client/Client/ui/ClientDetails

entities/Client/Document/ui/DocumentList
entities/Client/Document/ui/DocumentDetails

entities/Client/Event/ui/EventList
entities/Client/Event/ui/EventDetails

entities/Client/Note/ui/NoteList
entities/Client/Note/ui/NoteDetails

entities/Client/Check/ui/CheckList
entities/Client/Check/ui/CheckDetails

Или же нужно выносить всё обязательно на один уровень? Правда тогда , если есть сущность Staff и у него есть документы, события - то получим пересечение:

entities/Client/ui/ClientList
entities/Client/ui/ClientDetails

entities/Document/ui/DocumentList
entities/Document/ui/DocumentDetails

entities/Event/ui/EventList
entities/Event/ui/EventDetails

entities/Staff/ui/StaffList
entities/Staff/ui/StaffDetails

entities/Document/ui/DocumentList - intersection
entities/Document/ui/DocumentDetails - intersection

entities/Event/ui/EventList - intersection
entities/Event/ui/EventDetails - intersection

Допустим тот случай, когда переиспользовать EventList, DocumentList нельзя, так как они различаются для клиаента и стафа.

**Comment by illright** [comment, marked as answer]
Нет, подслайсы не рекомендуется делать в текущей версии методологии.

Смысл разделения на слайсы в том, чтоб максимально разграничить предметные области, упрощая таким образом поддержку кода. Тесно связанные сущности работают против этой цели.

Пересечение устраняется как раз независимостью сущности от других. То есть, если есть сущность Чек, она не знает, принадлежит она Пользователю или Сотруднику. Связь Чека и Пользователя происходит с помощью композиции на уровнях выше.

Иногда бывает так, что без связности сущностей не обойтись. Для этого есть RFC для кросс-импортов, который в будущем будет принят в основной методологии, его можно использовать уже сейчас:
https://github.com/feature-sliced/documentation/discussions/390#discussioncomment-5570073
